sdk/communication/AzureCommunicationChat/Source/Signaling/events/chat.swift

```diff
@@ -0,0 +1,676 @@
+        self.topic = topic
+    }
+}
+
+public class BaseChatEvent {
+    public var threadId: String
+    public var sender: CommunicationIdentifier?
+    public var senderDisplayName: String?
+    public var recipient: CommunicationIdentifier?
+
+    init(
+        threadId: String,
+        sender: CommunicationIdentifier?,
+        senderDisplayName: String? = nil,
+        recipient: CommunicationIdentifier?
+    ) {
+        self.threadId = threadId
+        self.sender = sender
+        self.senderDisplayName = senderDisplayName
+        self.recipient = recipient
+    }
+}
+
+public class BaseChatThreadEvent {
+    public var threadId: String
+    public var version: String
+
+    init(threadId: String, version: String) {
+        self.threadId = threadId
+        self.version = version
+    }
+}
+
+public class BaseChatMessageEvent: BaseChatEvent {
+    public var id: String
+    public var createdOn: Iso8601Date?
+    public var version: String
+    public var type: ChatMessageType
+
+    init(
+        threadId: String,
+        sender: CommunicationIdentifier?,
+        recipient: CommunicationIdentifier?,
+        id: String,
+        senderDisplayName: String? = nil,
+        createdOn: Iso8601Date? = nil,
+        version: String,
+        type: ChatMessageType
+    ) {
+        self.id = id
+        self.createdOn = createdOn
+        self.version = version
+        self.type = type
+        super.init(threadId: threadId, sender: sender, senderDisplayName: senderDisplayName, recipient: recipient)
+    }
+}
+
+public class ChatMessageReceivedEvent: BaseChatMessageEvent {
+    public var message: String
+
+    init(
+        threadId: String,
+        sender: CommunicationIdentifier?,
+        recipient: CommunicationIdentifier?,
+        id: String,
+        senderDisplayName: String? = nil,
+        createdOn: Iso8601Date?,
+        version: String,
+        type: ChatMessageType,
+        message: String
+    ) {
+        self.message = message
+        super.init(
+            threadId: threadId,
+            sender: sender,
+            recipient: recipient,
+            id: id,
+            senderDisplayName: senderDisplayName,
+            createdOn: createdOn,
+            version: version,
+            type: type
+        )
+    }
+
+    init(from request: TrouterRequest) throws {
+        guard let requestJsonData = request.body.data(using: .utf8) else {
+            throw AzureError.client("Unable to convert request body to Data.")
+        }
+
+        let messageReceivedPayload: MessageReceivedPayload = try JSONDecoder()
+            .decode(MessageReceivedPayload.self, from: requestJsonData)
+
+        self.message = messageReceivedPayload.messageBody
+        super.init(
+            threadId: messageReceivedPayload.groupId,
+            sender: TrouterEventUtil.getIdentifier(from: messageReceivedPayload.senderId),
+            recipient: TrouterEventUtil.getIdentifier(from: messageReceivedPayload.recipientId),
```

Comment:
@CHILIU-MSFT brought up a point that the recipientId we receive from Trouter is missing the prefix of "8:",  we need to double-check if it is the case, if so, we would be getting an UnknownIdentifier from here.

Reply:
Will update, was testing now and we do have to apply the workaround of hardcoding "8:", for GA will be fixed properly on backend